Model/AddressResolver.php

```diff
@@ -294,18 +298,30 @@ private function getCityId(AmwalAddressInterface $amwalAddress): ?string
             return null;
         }
 
+        $locale = $this->localeResolver->getLocale();
+
         $connection = $this->resourceConnection->getConnection();
         $tableName = $this->resourceConnection->getTableName('directory_country_region_city');
-
+        $localCityTableName = $this->resourceConnection->getTableName('directory_country_region_city_name');
         if (!$connection->isTableExists($tableName)) {
             return null;
         }
+        
+        $amwalCityName = $amwalAddress->getCity();
+        $condition = $connection->quoteInto('lng.locale = ?', $locale);
+        $name_match_condition = $connection->quoteInto('main_table.default_name = ?', $amwalCityName);
```

Comment:
The coding standards in Magento require the use of camelCase for variable names
```suggestion
        $nameMatchCondition = $connection->quoteInto('main_table.default_name = ?', $amwalCityName);
```